ideftbuild/nexjot_frontend
# Implement Save Functionality for Document Editor

This PR implements the save functionality for the document editing page, providing user feedback during the save process and preparing the codebase for future API integration.
**Key Changes**

* Save Button & Functionality: A "Save" button has been added to the document editor. Currently, it logs the updated document title and content in the browser console.
* User Feedback: The save button is disabled during the save process and displays "Saving...". Upon completion, a success or error message is displayed based on the outcome of the operation.
* Error Handling: The save logic is wrapped in a try-catch block to manage any failures during the save operation, ensuring the user is notified of errors without breaking the interface.


**Future API Integration:**

* Redux logic for document retrieval (introduced by a colleague) has been preserved and is currently commented out, ready for future API implementation.
* Mock data is used for document testing in the meantime.

**Optimistic UI Updates:** The UI is updated immediately when the save process begins, providing users with instant feedback while the operation completes.

**How to Test**

1. Navigate to the document editor and make changes to the document's title or content.
2. Click the "Save" button to trigger the save operation.
3. Confirm that:
    * The button is disabled during the save and shows "Saving...".
    * A success or error message is displayed upon completion.
    * The updated document is logged in the browser console.

Future Considerations
* Once the API is available, the handleSave function can be updated to perform actual save operations through a network request.
 * The commented Redux logic can be re-enabled to manage state more effectively when interacting with the backend.
